lib/rules/header.js

```diff
@@ -22,17 +22,34 @@ function excludeShebangs(comments) {
     });
 }
 
+// Returns either the first block comment or the first set of line comments that
+// are ONLY separated by a single newline. Note that this does not actually
+// check if they are at the start of the file since that is already checked by
+// hasHeader().
 function getLeadingComments(context, node) {
-    return node.body.length ?
-        context.getComments(node.body[0]).leading :
-        context.getComments(node).leading;
+    var all = excludeShebangs(context.getSourceCode().getAllComments(node.body.length ? node.body[0] : node));
+    if (all[0].type.toLowerCase() === "block") {
+        return [all[0]];
+    } else {
```

Comment:
Would be nice to skip else to avoid indents :D

Reply:
Ah yes. This PR evolved over time, should be possible to get rid of that now.